src/main/java/org/craftercms/engine/search/SiteAwareElasticsearchService.java

```diff
@@ -95,7 +101,10 @@ protected void updateFilters(final SearchRequest request) {
         if (auth != null && !(auth instanceof AnonymousAuthenticationToken) && isNotEmpty(auth.getAuthorities())) {
             logger.debug("Filtering search results for roles: {}", auth.getAuthorities());
             securityQuery.should(matchQuery(roleFieldName, auth.getAuthorities().stream()
-                            .map(Object::toString)
+                            .map(GrantedAuthority::getAuthority)
+                            .map(role -> role +  " " +
+                                    (startsWith(role, ROLE_PREFIX)? removeStart(role, ROLE_PREFIX)
```

Comment:
This confuses me: if the role has the `ROLE_` prefix, remove it. If not, append it to the end. What's the purpose? If it's complicated to explain ping me in Slack.

Reply:
This is to make it work without knowing if the value in `authorizedRoles` has the prefix or not. For example if the user has a role `ROLE_user` we will query for matches of both `ROLE_user` and `user`